### PR TITLE
Fix request params left unused

### LINKED ISSUE #30: GIF topic ignored in Tenor API code
# Bug Report
The request params in tenor API are left unused as the code does not add the much needed second `{params:}` argument.

## How to reproduce
1. Make a Tenor API call
2. None of the parameters get used
3. Tenor API errors out

## Intended behaviour
1. Make a Tenor API call
2. Parameters get passed
3. The API call is successful

### PR DESCRIPTION
**Pull request by zahatikoff**
# What Does This PR Fix
This PR adresses the Issue #30, where the Tenor API calls were not passing any of the parameters provided in the function, which lead to the said requests failing.

## Related Issues
Fixes #30

### CODE REVIEW

File: src/util/tenor-utils.ts [hunk 1]
```diff
@@ -170,20 +170,19 @@ export class TenorSingleton {
...
         * If the cache is full -> get a random one from cache
         */
-        const gifArray: string[] = TenorSingleton.cache.get(topic);
+        const cachedGifs: string[] = TenorSingleton.cache.get(topic);
 
-        //TODO: why the fuck should i compare stuff with !== and not !=, how strict is that?
         // eslint-disable-next-line no-magic-numbers
-        if (gifArray.length !== 0) {
-            return (gifArray[Math.floor(Math.random() * DEFAULT_RESPONSE_LIMIT)]);
+        if (cachedGifs?.length !== 0) {
```
Comment (zahatikoff) [inline src/util/tenor-utils.ts]: changed to the optional property as asked

File: src/util/tenor-utils.ts [hunk 2]
```diff
@@ -170,20 +170,19 @@ export class TenorSingleton {
...
+            return (cachedGifs[Math.floor(Math.random() * DEFAULT_RESPONSE_LIMIT)]);
         }
 
         /*
          * If there are no gifs in the cache for the specified topic -- fetch some 
          * from Tenor
          */
 
-        /*eslint-disable*/
+        /*eslint-disable camelcase*/
```
Comment (zahatikoff) [inline src/util/tenor-utils.ts:187]: specified the exact features needed to be disabled

File: src/util/tenor-utils.ts [hunk 3]
```diff
@@ -194,9 +193,10 @@ export class TenorSingleton {
             random: true,
             limit: DEFAULT_RESPONSE_LIMIT,
         };
-        /*eslint-enable*/
+        /*eslint-enable camelcase*/
         try {
-            const gifObjects: TenorResponseObject[] = (await this.client.get<TenorResponse>("/search")).data.results;
+            const gifObjects: TenorResponseObject[] = (await this.client.get<TenorResponse>("/search",{params:requestParams})).data.results;
```
Comment (zahatikoff) [inline src/util/tenor-utils.ts]: added back the necessary parameters field to the request

File: src/util/tenor-utils.ts [hunk 4]
```diff
@@ -170,20 +170,19 @@ export class TenorSingleton {
...
         * If the cache is full -> get a random one from cache
         */
-        const gifArray: string[] = TenorSingleton.cache.get(topic);
+        const cachedGifs: string[] = TenorSingleton.cache.get(topic);
 
-        //TODO: why the fuck should i compare stuff with !== and not !=, how strict is that?
         // eslint-disable-next-line no-magic-numbers
-        if (gifArray.length !== 0) {
-            return (gifArray[Math.floor(Math.random() * DEFAULT_RESPONSE_LIMIT)]);
+        if (cachedGifs!==null && cachedGifs?.length !== 0) {
```
Comment (MirageAegis) [inline src/util/tenor-utils.ts]: `cachedGifs?.length` is enough here

File: src/commands/action/pat.ts [hunk 5]
```diff
@@ -48,7 +48,7 @@ export const command: Command = {
         const tenor: TenorSingleton = TenorSingleton.getInstance();
         const patter: User = <User> ctx.member.user;
         const patted: User = <User> ctx.options.getUser("target");
-        const gif: string = await tenor.getGifs("anime+hug");
+        const gif: string = await tenor.getGifs("anime+headpat");
```
Comment (MirageAegis) [inline src/commands/action/pat.ts:51]: I fixed this overlooked mistake.

File: src/util/tenor-utils.ts [hunk 6]
```diff
@@ -130,85 +130,87 @@ type tenorMediaObject = {
...
         if (!this.instance) {
             this.instance = new TenorSingleton();
-            this.cache = new Map();
-            setInterval(() => {
-                this.cache.clear();
-            }, TWELVE_HOURS);
         }
+
         return this.instance;
     }
```
Comment (MirageAegis) [inline src/util/tenor-utils.ts:160]: I also cleaned up the `getInstance` function by creating the interval in the constructor.

File: src/util/tenor-utils.ts [hunk 7]
```diff
@@ -130,85 +130,87 @@ type tenorMediaObject = {
...
         } catch {
-            throw new TenorError;
+            throw new TenorError();
         }
+
+        // All gif URLs from the gif objects
+        const gifUrls: string[] = gifObjects.map(g => g.media_formats.gif.url);
+
+        this.cache.set(topic, gifUrls);        
+        return gifUrls[Math.floor(Math.random() * DEFAULT_RESPONSE_LIMIT)];
```
Comment (MirageAegis) [inline src/util/tenor-utils.ts:213]: Moved as many lines as possible out of the try block.

File: src/util/tenor-utils.ts [hunk 8]
```diff
@@ -130,85 +130,87 @@ type tenorMediaObject = {
...
+            gifObjects = (await this.client.get<TenorResponse>("/search", {
+                params:requestParams
+            })).data.results;
         } catch {
-            throw new TenorError;
+            throw new TenorError();
         }
+
+        // All gif URLs from the gif objects
+        const gifUrls: string[] = gifObjects.map(g => g.media_formats.gif.url);
```
Comment (MirageAegis) [inline src/util/tenor-utils.ts:210]: And condensed the array populating to a mapping function.

File: src/util/tenor-utils.ts [hunk 9]
```diff
@@ -130,85 +130,87 @@ type tenorMediaObject = {
 
 //
 
-const DEFAULT_RESPONSE_LIMIT = 100;
-const TWELVE_HOURS = 43_200_000;
+const DEFAULT_RESPONSE_LIMIT: number = 50;
```
Comment (MirageAegis) [inline src/util/tenor-utils.ts:133]: The `DEFAULT_RESPONSE_LIMIT` was also changed to reflect the limit put by Tenor.

**Review by MirageAegis — APPROVED** [review]
The commands seem to work now.

Btw, I took the liberty to clean up some things.

**Pull request merged**